auth/README.md

```diff
@@ -0,0 +1,69 @@
+# Auth Package
+
+Auth package is used for request authentication and authorization for user to service
+and service to service communication. Auth is used for scenarios that require stricter
+validations, and usage restrictions. `auth.Client` provided by the package provides seamless
+integration with presently written modules in the service framework. As a middleware, Auth
+provides additional restrictions on who can access a service, and who can be
+authenticated to access the service.
+The Auth package doesn't dictate how authentication and authorization should work, or which
+algorithm the security service should use, but only allows client integration with the service framework.
+
+## Auth calls
+SetAttribute:
+SetAttribute sets necessary request attributes for authentication. By setting attributes, security service can
+identify the service/user and grant certificate for further access.
+
+Authentication:
+Authentication API is called by calling entity to authenticate itself. The Authenticate call
+returns context, which must be populated by the back-end service with signed certificate that is valid for a time frame.
```

Comment:
- back-end --> backend
- Which backend service?
- Readers might want to know how long that time frame is